Amplify/Core/Support/TaskQueue.swift

```diff
@@ -25,17 +40,22 @@ public actor TaskQueue<Success> {
     /// TaskQueue serializes this work so that `doAsync1` is performed before `doAsync2`,
     /// which is performed before `doAsync3`.
     public func sync(block: @Sendable @escaping () async throws -> Success) async throws -> Success {
-        let currentTask: Task<Success, Error> = Task { [previousTask] in
-            _ = await previousTask?.result
-            return try await block()
+        try await withCheckedThrowingContinuation { continuation in
+            streamContinuation.yield {
+                do {
+                    let value = try await block()
+                    continuation.resume(returning: value)
+                
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
         }
-        previousTask = currentTask
-        return try await currentTask.value
     }
 
-    public nonisolated func async(block: @Sendable @escaping () async throws -> Success) rethrows {
-        Task {
-            try await sync(block: block)
+    public func async(block: @Sendable @escaping () async -> Success) {
```

Comment:
Has the removal of the `throws` keyword from the `block` closure's signature been intentional?

Reply:
Yes, because there is no way to throw an error from his function to a caller outside of this class as this function is async. `throws` keyword was indicating that something will catch/rethrow an eventual error but it will won't.

The more I think about it, the more I come to a conclusion that maybe this `throws` was not that bad, but we need to catch eventual errors ourselves inside this function and log them as error?

Reply:
Certainly. Yes, this asynchronous closure operates on a fire-and-forget basis. Errors occurring within it won't be propagated back to the invoker. 
Since this function is first designed as a public API, I'm particularly concerned about any changes to its signature.

Reply:
Exactly. 
Of course your concerns are understandable